cartoframes/data/dataset.py

```diff
@@ -504,7 +493,7 @@ def _get_geom_col_type(df):
         return None
 
     try:
-        geom = _decode_geom(_first_not_null_value(df, geom_col))
+        geom = decode_geometry(_first_value(df[geom_col]))
     except IndexError:
```

Comment:
Will this method raise an IndexError in any case? Maybe we could add a test case ;)

Reply:
This Exception won't be raised anymore. What case do you want to cover with tests?

Reply:
If first value returns `None` I guess `decode_geometry` will raise an exception, right? (not an IndexError) should we cover that case with a test?

Reply:
decode_geometry has an if internally, so the None case is ignored

Reply:
🤦‍♂ I missread the code, you are right!